base58/src/error.rs

```diff
@@ -40,73 +40,66 @@ impl From<BadByteError> for DecodeError {
     fn from(e: BadByteError) -> Self { Self::BadByte(e) }
 }
 
-/// An error that might occur during base58 decoding.
+/// An error occurred during base58 decoding (with checksum).
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[non_exhaustive]
-pub enum Error {
+pub enum DecodeCheckError {
     /// Decode error.
     Decode(DecodeError),
-    /// Checksum was not correct (expected, actual).
-    BadChecksum(u32, u32),
-    /// The length (in bytes) of the object was not correct.
-    ///
-    /// Note that if the length is excessively long the provided length may be an estimate (and the
-    /// checksum step may be skipped).
-    InvalidLength(usize),
-    /// Extended Key version byte(s) were not recognized.
-    InvalidExtendedKeyVersion([u8; 4]),
-    /// Address version byte were not recognized.
-    InvalidAddressVersion(u8),
-    /// Checked data was less than 4 bytes.
-    TooShort(usize),
-}
-
-impl fmt::Display for Error {
+    /// Checksum was not correct.
+    BadChecksum(BadChecksumError),
+    /// Checked data was too short.
+    TooShort(TooShortError),
```

Comment:
This is also invalid length so we should then just convert this one into `InvalidLength` downstream.

Reply:
Conceptually you are correct but since I've argued above that we should not have the `InvalidLength` then I believe this comment is mute.